### DISCUSSION TITLE
sets function

### DISCUSSION THREAD

**Discussion by neokofg** [laravel/framework#49922] [Ideas]
i think it would be great to add, sets function, it like casts, but it sets attributes when saving/creating/updating for example, column phone, you can add function to this column in sets function, for example to add + in start of value, so then we don't need to add set and get methods inside the model
```php
protected function sets(): array
{
    return [
        'email_verified_at' => $this->someSortOfFunc(),
        'password' => fn () => some func,
    ];
}
```

**Comment by henzeb** [comment]
for simple solutions we have $attributes you can set, you can also do it in the constructor, nothing wrong with that. There's the boot method. 

But if we are going to do this, I suggest ditch the `sets` name and call it `defaultAttributes`. Then it says what it does.
